program.c

```diff
@@ -0,0 +1,111 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+int is_digit(char digit)
+{
+    if (digit >= '0' && digit <= '9')
+        return 1;
+    else
+        return 0;
+}
+int is_operator(char operator)
+{
+    if (operator== '+' || operator== '-' || operator== '/' || operator== '*')
+        return 1;
+    else
+        return 0;
+}
+long long int apply_operation(long long int number1, long long int number2, char operator)
+{
+    if (operator== '+')
+        return number1 + number2;
+    if (operator== '-')
+        return number1 - number2;
+    if (operator== '*')
+        return number1 * number2;
+    if (operator== '/')
+    {
+        if (number2 == 0)
+        {
+            printf("Error: Division by zero.\n");
+            exit(0);
+        }
+        return number1 / number2;
+    }
+    return 0;
+}
+long long int evaluate_expression(char *expression)

```

Comment:
Do error handling for the edge cases like 5+, -5*-2, 3/(5-5), empty input, -5*2.